### ISSUE TITLE
git error - "detected dubious ownership in repository"

### ISSUE THREAD

**Issue by HughMacdonald** [dorny/test-reporter#226]
I've got 2 repositories that use the same container:

```
    runs-on: ubuntu-latest
    container: node:latest
```

One of them works fine, and the other has this error:

```
  /usr/bin/git ls-files -z
  fatal: detected dubious ownership in repository at '/__w/MyRepository/MyRepository'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /__w/MyRepository/MyRepository
```

The one that comes up with this error also does some additional checking out of a dependency:
```
      - name: Checkout MyRepository
        uses: actions/checkout@v3
        with:
          repository: MyOrganisation/MyRepository
          token: ${{ secrets.GH_PAT }}
          path: dependencies/MyRepository
          ref: develop
```

Any idea what could be causing this? The other repository (that doesn't have the dependency) doesn't have this issue.

**Comment by MichaelVoelkel** [comment]
I have no idea what could be causing this but in my fork I just ignore if this check fails. It will still show the error in the log but it will just assume there are no files. I tested it and it unblocks me: https://github.com/MichaelVoelkel/test-reporter
Commit: https://github.com/MichaelVoelkel/test-reporter/commit/17034361a915aba54df961fb449958b0f2c07087

**Comment by sbouchexbellomie-Philips** [comment]
We are having the same issue when running on self-hosted runners on Ubuntu 24.04. It's working fine on Ubuntu 22.04.

checkout action is having the same issue [https://github.com/actions/checkout/issues/2031](url)

Workaround: git config --global --add safe.directory "$GITHUB_WORKSPACE"

**Comment by jozefizso** [comment]
I am closing this report as it is out of scope of this project.
